lib/rubocop/cop/rspec/capybara/specific_actions.rb

```diff
@@ -0,0 +1,136 @@
+        #   find('a').click
+        #   find('button.cls').click
+        #   find('a', exact_text: 'foo').click
+        #   find('div button').click
+        #
+        #   # good
+        #   click_link
+        #   click_button(class: 'cls')
+        #   click_link(exact_text: 'foo')
+        #   find('div').click_button
+        #
+        class SpecificActions < Base
+          MSG = "Prefer `%<good_action>s` over `find('%<selector>s').click`."
+          RESTRICT_ON_SEND = %i[click].freeze
+          SPECIFIC_ACTION = {
+            'button' => 'button',
+            'a' => 'link'
+          }.freeze
+
+          # @!method first_argument(node)
+          def_node_matcher :first_argument, <<-PATTERN
+            (send _ :find (str $_) ...)
+          PATTERN
+
+          # @!method option?(node)
+          def_node_search :option?, <<-PATTERN
+            (pair (sym %) _)
+          PATTERN
+
+          def on_send(node)
+            return unless (receiver = receiver(node))
+
+            first_argument(receiver) do |arg|
+              return unless supported_selector?(arg)
+              return unless (selector = last_selector(arg)[/^\w+/, 0])
+              return unless (action = specific_action(selector))
+              next unless specific_action_option?(receiver, arg, action)
+              next unless specific_action_pseudo_classes?(arg)
+
+              range = offense_range(node, receiver)
+              add_offense(range, message: message(action, selector))
+            end
+          end
+
+          def receiver(node)
+            return unless (receiver = node.receiver)
+
+            receiver if receiver.method?(:find)
+          end
+
+          def specific_action(selector)
+            SPECIFIC_ACTION[last_selector(selector)]
+          end
+
+          def specific_action_option?(node, arg, action)
+            attrs = CssSelector.attributes(arg).keys
+            return false unless replaceable_action?(node, action, attrs)
+
+            attrs.all? do |attr|
+              CssSelector.specific_options?(action, attr)
+            end
+          end
+
+          def specific_action_pseudo_classes?(arg)
+            CssSelector.pseudo_classes(arg).all? do |pseudo_class|
+              replaceable_pseudo_class?(pseudo_class, arg)
+            end
+          end
+
+          def replaceable_pseudo_class?(pseudo_class, arg)
+            unless CssSelector.specific_pesudo_classes?(pseudo_class)
+              return false
+            end
+
+            case pseudo_class
+            when 'not()' then replaceable_pseudo_class_not?(arg)
+            else true
+            end
+          end
+
+          def replaceable_pseudo_class_not?(arg)
+            arg.scan(/not\(.*?\)/).all? do |not_arg|
+              CssSelector.attributes(not_arg).values.all? do |v|
+                v.is_a?(TrueClass) || v.is_a?(FalseClass)
+              end
+            end
+          end
+
+          def replaceable_action?(node, action, attrs)
```

Comment:
perhaps we need to start extracting common parts of the cops

Reply:
Indeed it is. I extracting common parts of the cops. These seemed like different methods from CssSelector, so we carved them out into a separate module as CapybaraHelp.

Reply:
We have it as a separate commit, but if it needs to be squashed, please point it out.